script.js

```diff
@@ -153,40 +172,22 @@ button.addEventListener('click', function(){
                     var newDateFormat = new Date(alltimeValue[i] * 1000);
                     alltimeValue[i] = newDateFormat.getDate()+'/'+newDateFormat.getMonth()+1+'/'+newDateFormat.getFullYear()+' - '+newDateFormat.getHours()+'h';
                 };
-                var datasetsoptional = new Object();
-                let listdatasets = [];
-                let listoflabels = [];
-                let listofdata =[];
+                debugger;
+
                 for (let a of checkedElements){
                     a = a.value
-                    datasetsoptional.label = a + data['hourly_units'][`${a}`];
-                    listoflabels.push(datasetsoptional.label);
-                    datasetsoptional.data  = data['hourly'][`${a}`];
-                    listofdata.push(datasetsoptional.data);
-                    var jsonString= JSON.stringify(datasetsoptional);
-                    listdatasets.push(jsonString)
+                    debugger;
+                    plotchar(data, a, alltimeValue);
```

Comment:
When called again (i.e. if I press the Submit button again), the execution fails because of this kind of error:
`Error: Canvas is already in use. Chart with ID '0' must be destroyed before the canvas with ID 'relativehumidity_2m_chart' can be reused.`

Reply:
Good, works now - however when submitting again, I get NaNs instead of time values in the charts.

Reply:
Working now, however seems to be too complicated - and in general, overwriting stored values (like in the cache) should be avoided as it may have side effects...

I would prefer using another variable for formatted dates, then the problem is gone (and the formatted dates can be stored also in the cache if needed).